apps/dapp/src/components/Pages/Core/NewUI/DefendModal.tsx

```diff
@@ -0,0 +1,141 @@
+import { Popover } from 'components/Popover';
+import styled from 'styled-components';
+import { Button } from 'components/Button/Button';
+import { useWallet } from 'providers/WalletProvider';
+import { useState } from 'react';
+import { formatBigNumber, formatTemple, getBigNumberFromString } from 'components/Vault/utils';
+import { ZERO } from 'utils/bigNumber';
+import { BigNumber } from 'ethers';
+import { CryptoSelector, CryptoValue, Input } from './HomeInput';
+import { TICKER_SYMBOL } from 'enums/ticker-symbol';
+import { formatToken } from 'utils/formatter';
+import env from 'constants/env';
+import { TempleStableAMMRouter__factory } from 'types/typechain';
+
+interface IProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const { DAI, TEMPLE_TOKEN } = TICKER_SYMBOL;
+
+const inputConfig: CryptoValue = {
+  kind: 'value',
+  value: `${TEMPLE_TOKEN}`,
+};
+
+const outputConfig: CryptoSelector = {
+  kind: 'select',
+  cryptoOptions: [DAI],
+  onCryptoChange: () => {},
+  selected: DAI,
+};
+
+export const DefendModal: React.FC<IProps> = ({ isOpen, onClose }) => {
+  const { wallet, signer, balance, updateBalance } = useWallet();
+  const [inputValue, setInputValue] = useState('');
+
+  const sell = async () => {
+    // Initialize router contract
+    if (!signer || !wallet) return;
+    const router = new TempleStableAMMRouter__factory(signer).attach(env.contracts.templeV2Router);
+
+    // Build swap parameters
+    const fraxAddress = env.contracts.frax; // Use FRAX address even though Defend returns DAI
+    const deadline = 20;
+    const deadlineTimestamp = Math.floor(Date.now() / 1000) + deadline * 60;
+    const amountIn = getBigNumberFromString(inputValue, 18);
+    const amountOut = amountIn.mul(10_000).div(8_000); // $0.80 per TEMPLE
```

Comment:
Probably best to call `router.swapExactTempleForStableQuote()`? That way if we change defend, this doesn't need to change.

This quote returns a `quote.priceBelowIV` so you know if it's using defend (and so you can throw an exception if it's not using defend).

Also slippage doesn't apply when using defend

Reply:
added on-chain quote: [e08b58f](https://github.com/TempleDAO/temple/pull/779/commits/e08b58f5eb40123d1fbacd189c1ab0b0b6b342e0)

will keep the slippage calc even if the swap will go to Defend just for simple logic